docs/src/modules/components/MarkdownDocsV2.js

```diff
@@ -0,0 +1,256 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { useRouter } from 'next/router';
+import kebabCase from 'lodash/kebabCase';
+import { useTheme } from '@mui/system';
+import { exactProp } from '@mui/utils';
+import Box from '@mui/material/Box';
+import { CssVarsProvider, useColorScheme } from '@mui/joy/styles';
+import ComponentsApiContent from 'docs/src/modules/components/ComponentsApiContent';
+import HooksApiContent from 'docs/src/modules/components/HooksApiContent';
+import { getTranslatedHeader as getHookTranslatedHeader } from 'docs/src/modules/components/HookApiPage';
+import { getTranslatedHeader as getComponentTranslatedHeader } from 'docs/src/modules/components/ApiPage';
+import MarkdownElement from 'docs/src/modules/components/MarkdownElementV2';
+import { pathnameToLanguage } from 'docs/src/modules/utils/helpers';
+import AppLayoutDocs from 'docs/src/modules/components/AppLayoutDocsWithoutAppFrame';
+import { useTranslate, useUserLanguage } from 'docs/src/modules/utils/i18n';
+import BrandingProvider from 'docs/src/BrandingProvider';
+import Ad from 'docs/src/modules/components/Ad';
+import AdGuest from 'docs/src/modules/components/AdGuest';
+
+function JoyModeObserver({ mode }) {
+  const { setMode } = useColorScheme();
+  React.useEffect(() => {
+    setMode(mode);
+  }, [mode, setMode]);
+  return null;
+}
+
+JoyModeObserver.propTypes = {
+  mode: PropTypes.oneOf(['light', 'dark']),
+};
+
+export default function MarkdownDocs(props) {
```

Comment:
Once we migrate all docs to use tabs, the `MarkdownDocs.js` will be replaced by `MarkdownDocsV2.js`. I isolated the two, so that we won't accidently break something in the other pages.